### PR TITLE
Added code to address #210.

### PR DESCRIPTION
**Pull request by umbernhard**
* implemented a flag for the port
* change default dynamically based on whether or not we're using TLS

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -255,10 +277,17 @@ func init() {
 	config.Encoding = encoding
 
 	// Validate port
-	if portFlag > 65535 {
-		zlog.Fatal("Port", portFlag, "out of range")
+	if portFlag.value > 65535 {
```
Comment (paul-pearce) [inline main.go]: It's impossible to ever hit this now, as the cast has already occured.

File: main.go [hunk 2]
```diff
@@ -32,6 +33,25 @@ import (
...
 
+type port struct {
+	value uint16
+	set   bool
+}
+
+func (pf *port) Set(x string) error {
+	value, err := strconv.Atoi(x)
+	if err != nil {
+		zlog.Fatal("Port", x, "is invalid")
```
Comment (paul-pearce) [inline main.go]: Missing spaces in the error msg

File: main.go [hunk 3]
```diff
@@ -32,6 +33,25 @@ import (
 	"github.com/zmap/zgrab/ztools/ztls"
 )
 
+type port struct {
+	value uint16
+	set   bool
```
Comment (paul-pearce) [inline main.go]: It seems heavy handed to keep a bool around for the lifetime of the port just to identify if it is set. You could accomplish this with a sentential value, a ptr, or separate data structure that only lives the lifetime of the configuration.

**Comment by paul-pearce** [comment]
I'm not a maintainer on this so obviously defer to @dadrian or @zakird, but my gut reaction would be to revert back to uint for the port, and use 0 as a sentinel value (treat `port ==0` as `port.set == false` in this code). It's not perfect, as if a user passes in port 0 it will behave as though they passed in nothing. But if the user passes in 0 they are going to have bigger problems on their hands.

File: main.go (hunk 2 above)
Comment (umbernhard) [inline main.go]: I thought I didn't need them? If I do we should fix line 281 too.

**Comment by dadrian** [comment]
Why don't we just require `--port` be passed in? This doesn't seem particularly worth fixing right now, without doing a much larger overhaul of how ZGrab does modules / flags in general.

**Comment by zakird** [comment]
I agree with @dadrian here.

**Pull request closed without merging**